packages/aws-cdk/lib/api/cxapp/cloud-executable.ts

```diff
@@ -66,9 +66,16 @@ export class CloudExecutable {
     while (true) {
       const assembly = await this.props.synthesizer(this.props.sdkProvider, this.props.configuration);
 
-      if (assembly.manifest.missing) {
+      if (assembly.manifest.missing && assembly.manifest.missing.length > 0) {
         const missingKeys = missingContextKeys(assembly.manifest.missing);
 
+        if (!this.canLookup) {
+          throw new Error(
+            'Context lookups have been disabled. '
+            + 'Make sure all necessary context is already in \'cdk.context.json\' by running \'cdk synth\' on a machine with sufficient AWS credentials and committing the result. '
+            + `Missing context keys: '${Array.from(missingKeys)}'`);
```

Comment:
This makes bold assumptions about how an array will be stringified... I wouldn't recommend...

```suggestion
            + `Missing context keys: '${Array.from(missingKeys).join(', ')}'`);
```